exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -252,13 +255,41 @@ func (prwe *PrwExporter) handleSummaryMetric(tsMap map[string]*prompb.TimeSeries
 }
 
 // export sends a Snappy-compressed WriteRequest containing TimeSeries to a remote write endpoint in order
-func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) error {
-	// Calls the helper function to convert the TsMap to the desired format
-	req, err := wrapTimeSeries(tsMap)
+func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) []error {
+	var errs []error
+	// Calls the helper function to convert and batch the TsMap to the desired format
+	requests, err := batchTimeSeries(tsMap)
 	if err != nil {
-		return consumererror.Permanent(err)
+		errs = append(errs, consumererror.Permanent(err))
+		return errs
 	}
 
+	var mu sync.Mutex
+	numOfRequests := len(requests)
+
+	for i := 0; i < numOfRequests; i += maxConcurrentRequests {
```

Comment:
This will wait for the slowest request to be completed before starting to consume the next. We can use a buffered channel instead. 

An example of this pattern is https://github.com/google/gops/blob/master/goprocess/gp.go#L55

Not sure if it's readable enough, let me know if you need help.

Reply:
Ohh, I see! I'll try that out